crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -266,6 +266,7 @@ impl TEdgeComponent for CumulocityMapper {
             Some(AvailabilityBuilder::new(
                 AvailabilityConfig::try_new(&tedge_config, c8y_profile)?,
                 &mut c8y_mapper_actor,
+                &mut mqtt_actor,
```

Comment:
Breaking the MQTT source and sink was definitely a good idea. But, I feel we need to flip the source and sink. It might be okay for the availability actor to receive the messages directly from the MQTT actor, but the output must be routed through the c8y actor for the reason described in https://github.com/thin-edge/thin-edge.io/pull/3289/files#r1877418069